build-tools/packages/build-tools/src/fluidBuild/tasks/taskFactory.ts

```diff
@@ -75,7 +72,32 @@ const executableToLeafTask: {
 	// pipeline then this mapping will have to be updated.
 	"renamer": RenameTypesTask,
 	"flub rename-types": RenameTypesTask,
-};
+} as const;
+
+/**
+ * Given a command executable, attempts to find a matching TaskHandler that will handle the task. If one is found, it is
+ * returned; otherwise, it returns `UnknownLeafTask` as the default handler.
+ *
+ * The built-in executableToLeafTask constant is checked first, followed by any DeclarativeTasks that are defined in the
+ * fluid-build config.
+ *
+ * @param executable The command executable to find a matching task handler for.
+ * @returns A TaskHandler for the task, if found. Otherwise `UnknownLeafTask` as the default handler.
+ */
+function getTaskForExecutable(executable: string, context: BuildContext): TaskHandler {
+	const found: TaskHandler | undefined = executableToLeafTask[executable];
+	if (found === undefined) {
+		const config = context.fluidBuildConfig;
+		const declarativeTasks = config?.declarativeTasks;
+		const taskMatch = declarativeTasks?.[executable];
+		if (taskMatch !== undefined) {
+			return createDeclarativeTaskHandler(taskMatch);
+		}
+	}
+
+	// If no handler is found, return the UnknownLeafTask as the default handler.
+	return found ?? UnknownLeafTask;
```

Comment:
If we fail the build, then it means no one can add a script to the build that doesn't have a task handler. That would
mean that all tasks would be cacheable (nice) but it would also mean there would have to be a task handler for any
new build tool we start using - depcheck, ts2esm, biome, etc. While we would generally want a task handler eventually, the loose
coupling here allows folks that aren't build experts to add tasks pretty easily, and still get the scheduling benefits
of fluid-build.

Reply:
UnknownLeafTask just execs the command, but doesn't cache anything, nor does it support caching.